### PR TITLE
FastSim number of photoelectrons in HF simulation

### PR DESCRIPTION
**Pull request by sbein**
The parameter ProbMax in FastSim HF simulation is updated to be consistent with the change made in FullSim:

https://github.com/cms-sw/cmssw/pull/25155

This fixes an issue related to the HF jet response discussed in the Feb 4 simulation meeting:

https://indico.cern.ch/event/1007716/contributions/4231234/attachments/2189371/3700030/FastSimBugsFeb2021b.pdf

This also fixes a discrepancy in the ttbar MET distribution introduced in CMSSW_10_4_0_pre3. 
#### PR validation:

the update has been tested in the head release of C11 and also in the UL release CMSSW_10_6_20 (plots of jet response and MET attached). This change needs to be backported to CMSSW_10_6. 

[fastvsfull_forwardJetResponse_CMSSW_10_6_20_fix.pdf](https://github.com/cms-sw/cmssw/files/6074744/fastvsfull_forwardJetResponse_CMSSW_10_6_20_fix.pdf)
[fastvsfull_forwardJetResponse_CMSSW_11_X.pdf](https://github.com/cms-sw/cmssw/files/6074745/fastvsfull_forwardJetResponse_CMSSW_11_X.pdf)
[fastvsfull_MET_CMSSW_10_6_20_fix.pdf](https://github.com/cms-sw/cmssw/files/6074746/fastvsfull_MET_CMSSW_10_6_20_fix.pdf)
[fastvsfull_MET_CMSSW_11_X.pdf](https://github.com/cms-sw/cmssw/files/6074747/fastvsfull_MET_CMSSW_11_X.pdf)

### CODE REVIEW

File: FastSimulation/Calorimetry/python/Calorimetry_cff.py [hunk 1]
```diff
@@ -262,7 +262,7 @@
             timeShiftHF = cms.vdouble(50.7, 52.5, 52.9, 53.9, 54.5, 55.1, 55.1, 55.7, 55.9, 56.1, 56.1, 56.1, 56.5),
             ),
         HFShower           = cms.PSet(
-            ProbMax          = cms.double(1.0),
+            ProbMax          = cms.double(0.5),
```
Comment (mariadalfonso) [inline FastSimulation/Calorimetry/python/Calorimetry_cff.py:265]: this is not ideal.
the ProbMax should stay 1.0 for Run1 and then change to 0.5 from the Run2 and beyond.
https://github.com/cms-sw/cmssw/pull/25155/files#diff-5ce5dd838a17416becd944c6910f93ea61928b0bc0ed63ddfb9b8ad19347d974R534 

Better import directly the parameter from FullSim config

**Comment by abdoulline** [comment]
@sbein    Sam, 
are those HF samplingFactors actually used  ? - 
https://cmssdt.cern.ch/lxr/source/FastSimulation/Calorimetry/python/Calorimetry_cff.py#0250
E.g. 
https://cmssdt.cern.ch/lxr/source/FastSimulation/Calorimetry/src/CalorimetryManager.cc#1133

NB: for FullSim we have (as specified in https://github.com/cms-sw/cmssw/pull/25155) slightly different values now...

File: FastSimulation/Calorimetry/python/Calorimetry_cff.py (hunk 1 above)
Comment (sbein) [reply FastSimulation/Calorimetry/python/Calorimetry_cff.py]: Ok thanks for catching this and providing the solution.

**Comment by sbein** [comment]
@abdoulline 
I believe the values
https://cmssdt.cern.ch/lxr/source/FastSimulation/Calorimetry/python/Calorimetry_cff.py#0250

are not used at all because the default value of  useShowerLibrary is True:
https://cmssdt.cern.ch/lxr/source/FastSimulation/Calorimetry/python/Calorimetry_cff.py#0270

but if useShowerLibrary were to be set to false, they seem like they would be used.

**Comment by abdoulline** [comment]
> @abdoulline
> I believe the values
> https://cmssdt.cern.ch/lxr/source/FastSimulation/Calorimetry/python/Calorimetry_cff.py#0250
> 
> are not used at all because the default value of useShowerLibrary is True:
> https://cmssdt.cern.ch/lxr/source/FastSimulation/Calorimetry/python/Calorimetry_cff.py#0270
> 
> but if useShowerLibrary were to be set to false, they seem like they would be used.

OK, thanks.

**Comment by civanch** [comment]
please test

**Comment by sbein** [comment]
+1

**Comment by qliphy** [comment]
@sbein  Just to confirm,  workflow 135.4 shows many difference and this is expected as 135.4 is a fastsim one, right?

**Comment by abdoulline** [comment]
@qliphy   I take the liberty to answer -  this is  FastSim: 
--fast  --era Run2_2016

https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-69c7b8/13315/runTheMatrix-results/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS/ZEE_13TeV_TuneCUETP8M1_cfi_GEN_SIM_RECOBEFMIX_DIGI_L1_DIGI2RAW_L1Reco_RECO_EI_VALIDATION_DQM.py

**Comment by qliphy** [comment]
+1

**Pull request merged**